in_toto/runlib.py

```
@@ -570,6 +589,10 @@ def in_toto_run(name, material_list, product_list, link_cmd_args,
   if signature:
     signing_keyid = signature["keyid"]
     filename = FILENAME_FORMAT.format(step_name=name, keyid=signing_keyid)
+
+    if metadata_directory is not None:
```

Could you add a:
```python
  if metadata_directory:
    securesystemslib.formats.PATH_SCHEMA.check_match(metadata_directory)
```
on top of the function body to fail early?

---

Same goes for `in_toto_record_stop`